src/no/runsafe/warpdrive/database/WarpRepository.java

```diff
@@ -164,37 +161,60 @@ private ILocation GetWarp(IPlayer owner, String name, boolean publicWarp)
 				name
 			);
 		else
-		{
-			String ownerId = "";
-			if (owner != null)
-				ownerId = owner.getUniqueId().toString();
 			location = database.queryLocation(
-				"SELECT world, x, y, z, yaw, pitch FROM warpdrive_locations WHERE name=? AND `public`=0 AND creator_id=?",
-				name, ownerId
+				"SELECT world, x, y, z, yaw, pitch FROM warpdrive_locations WHERE name=? AND `public`=0 AND creator=?",
+				name, owner
 			);
-		}
 
 		return cache.Cache(key, location);
 	}
 
+	/**
+	 * Checks if a warp is in the database.
+	 * @param owner Warp creator. Only needs to be valid if the warp is private.
+	 * @param name Warp name.
+	 * @param publicWarp True if it's a public warp.
+	 * @return True if the warp exists even if it has an invalid location.
+	 */
+	private boolean doesWarpExist(IPlayer owner, String name, boolean publicWarp)

```

Comment:
Again, just check if `owner` is set to `null` rather than having a separate bool, I'd say.